lib/player.js

```diff
@@ -5276,11 +5287,117 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     selectSrcEqualsMode();
   }
 
+  /**
+   * Select an audio track compatible with the current video track.
+   * If the player has not loaded any content, this will be a no-op.
+   *
+   * @param {shaka.extern.AudioTrack} audioTrack
+   * @param {number=} safeMargin Optional amount of buffer (in seconds) to
+   *   retain when clearing the buffer. Useful for switching quickly
+   *   without causing a buffering event. Defaults to 0 if not provided. Can
+   *   cause hiccups on some browsers if chosen too small, e.g. The amount of
+   *   two segments is a fair minimum to consider as safeMargin value.
+   * @export
+   */
+  selectAudioTrack(audioTrack, safeMargin = 0) {
+    const variants = this.getVariantTracks();
+    if (!variants.length) {
+      return;
+    }
+    const active = variants.find((t) => t.active);
+    if (!active) {
+      return;
+    }
+    const getRolesString = (roles) => {
+      if (!roles) {
+        return '';
+      }
+      return roles.join(',');
+    };
+    const validVariant = variants.find((t) => {
+      return t.videoId === active.videoId &&
+          t.language == audioTrack.language &&
+          t.label == audioTrack.label &&
+          t.audioMimeType == audioTrack.mimeType &&
+          t.audioCodec == audioTrack.codecs &&
+          t.primary == audioTrack.primary &&
+          getRolesString(t.audioRoles) == getRolesString(audioTrack.roles) &&
+          t.accessibilityPurpose == audioTrack.accessibilityPurpose &&
+          t.channelsCount == audioTrack.channelsCount &&
+          t.audioSamplingRate == audioTrack.audioSamplingRate &&
+          t.spatialAudio == audioTrack.spatialAudio &&
+          t.originalLanguage == audioTrack.originalLanguage;
+    });
+    if (validVariant && !validVariant.active) {
+      this.selectVariantTrack(validVariant,
+          /* clearBuffer= */ true, safeMargin);
+    }
+  }
+
+
+  /**
+   * Return a list of audio tracks compatible with the current video track.
+   *
+   * @return {!Array<shaka.extern.AudioTrack>}
+   * @export
+   */
+  getAudioTracks() {
+    const variants = this.getVariantTracks();
```

Comment:
Performance wise, on every call to this method we will create variant track objects. Can we operate on internal Variant objects instead?

Reply:
This unifies the src= mode and the MSE mode, so I think it's better this way to avoid duplicating a lot of code.